File: apps/ocr/src/ocr_service/api/health.py
"""Health and build info endpoints."""

from __future__ import annotations

import time

from fastapi import APIRouter

from ..schemas import BuildInfoResponse, HealthResponse
from ..settings import Settings


def get_router(settings: Settings) -> APIRouter:
    start_time = time.time()
    router = APIRouter()

    @router.get("/health", response_model=HealthResponse)
    async def health_check():
        return HealthResponse(
            status="healthy",
            service="ocr-service",
            version=settings.version,
            uptime_seconds=round(time.time() - start_time, 2),
        )

    @router.get("/build-info", response_model=BuildInfoResponse)
    async def build_info():
        return BuildInfoResponse(
            service="ocr-service",
            version=settings.version,
            git_sha=settings.git_sha,
            build_time=settings.build_time,
        )

    return router

File: apps/fhe/src/routes/health.rs
//! Health and build-info endpoints.

use axum::Json;
use serde::Serialize;

#[derive(Serialize)]
pub struct HealthResponse {
    status: String,
    service: String,
}

pub async fn health() -> Json<HealthResponse> {
    Json(HealthResponse {
        status: "ok".to_string(),
        service: "fhe-service".to_string(),
    })
}

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BuildInfoResponse {
    service: String,
    version: String,
    git_sha: String,
    build_time: String,
}

/// Build info endpoint for deployment verification.
/// Values are embedded at compile time via build.rs.
pub async fn build_info() -> Json<BuildInfoResponse> {
    Json(BuildInfoResponse {
        service: "fhe-service".to_string(),
        version: env!("CARGO_PKG_VERSION").to_string(),
        git_sha: env!("GIT_SHA").to_string(),
        build_time: env!("BUILD_TIME").to_string(),
    })
}
